### PR TITLE
ARCCodeMotion: add debug tracing.

### PR DESCRIPTION
**Pull request by atrick**

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci test.

File: lib/SILOptimizer/Transforms/ARCCodeMotion.cpp [hunk 1]
```diff
@@ -180,6 +180,10 @@ class CodeMotionContext {
   /// we compute the genset and killset.
   llvm::SmallPtrSet<SILBasicBlock *, 8> InterestBlocks;
 
+#ifndef NDEBUG
+  Optional<SILPrintContext> printCtx;
```
Comment (eeckstein) [inline lib/SILOptimizer/Transforms/ARCCodeMotion.cpp]: What's the purpose of printCtx? To set the SortedSIL flag?

**Comment by atrick** [comment]
@swift-ci smoke test.

File: lib/SILOptimizer/Transforms/ARCCodeMotion.cpp (hunk 1 above)
Comment (atrick) [reply lib/SILOptimizer/Transforms/ARCCodeMotion.cpp]: I added a comment:
```
+  // SILPrintContext is used to print block IDs in RPO order.
+  // It is optional so only the final insertion point interference is printed.
   Optional<SILPrintContext> printCtx;
```

It looks like I copied this from RLE's debug output.

Comment (eeckstein) [reply lib/SILOptimizer/Transforms/ARCCodeMotion.cpp]: ok, makes sense. Thanks

**Review by eeckstein — APPROVED** [review]
lgtm

**Pull request merged**